Formula/signal-cli.rb

```diff
@@ -1,17 +1,84 @@
 class SignalCli < Formula
   desc "CLI and dbus interface for WhisperSystems/libsignal-service-java"
   homepage "https://github.com/AsamK/signal-cli"
-  url "https://github.com/AsamK/signal-cli/releases/download/v0.7.4/signal-cli-0.7.4.tar.gz"
-  sha256 "d690871e411ed897c30a4b17e6dece2fe8f145590f4b7186a13e6b4b4812bba1"
+  url "https://github.com/AsamK/signal-cli/releases/download/v0.8.1/signal-cli-0.8.1.tar.gz"
+  sha256 "2ead51489d5521ae8c1538936c6a8dcbb22a96227019eb9e826ff5f9146dbbe2"
   license "GPL-3.0-or-later"
 
-  bottle :unneeded
+  depends_on "gradle" => :build
+  depends_on "protobuf" => :build
+  # the libsignal-client build targets a specific rustc (nightly-2020-11-09)
+  # which doesn't automatically happen if we use brew-installed rust. rustup-init
+  # allows us to use a toolchain that lives in HOMEBREW_CACHE
+  depends_on "rustup-init" => :build
 
   depends_on "openjdk"
 
+  resource "libsignal-client" do
+    # per https://github.com/AsamK/signal-cli/wiki/Provide-native-lib-for-libsignal#libsignal-client
+    # we want the specific libsignal-client version from 'signal-cli-0.8.1/lib/signal-client-XXXX-X.X.X.jar'
+    version "java-0.2.3"
+    url "https://github.com/signalapp/libsignal-client/archive/refs/tags/#{version}.tar.gz"
+    sha256 "730c1dc113da5227920716656d8f888e1af167208e095a8cac3de9c0d83890c4"
+  end
+
+  resource "libzkgroup" do
+    # per https://github.com/AsamK/signal-cli/wiki/Provide-native-lib-for-libsignal#libzkgroup
+    # we want the latest release version
+    url "https://github.com/signalapp/zkgroup/archive/refs/tags/v0.7.2.tar.gz"
+    sha256 "fdd03bbf584533963d1be40ab238d4e6199b379e8112f6aaf5cd9493b7f1fb47"
+  end
```

Comment:
Can't these be formulae?

Reply:
Would that be useful if they're not needed in any other formulae?

Reply:
Yeah, if it's independent software we should try and make it a formula. That way it can be checked for updates and everything.

Reply:
I guess `libzkgroup` can be its own formula, since we always want the latest version. `libsignal-client` seems to have specific version requirements, but I don't mind having a formula as the latest version to see if it works.

Though I guess it would be annoying to have a `libsignal-client` formula that isn't actually useful for `signal-cli`.

Reply:
Hm. Let me see what things look like if I just try to entirely build signal-cli from source?

Right now I'm just following the upstream [recommended install steps](https://github.com/AsamK/signal-cli#installation) which unfortunately is basically "download the prebuilt binary package, and if you're not on linux-x86_64 you also have to rebuild/replace the libsignal & libzkgroup libraries inside that prebuilt package."

(It looks like [0.8.0](https://github.com/AsamK/signal-cli/blob/master/CHANGELOG.md#080---2021-02-14) added the libsignal-client dependency, and [0.7.0](https://github.com/AsamK/signal-cli/blob/master/CHANGELOG.md#070---2020-12-15) added the libzkgroup dependency, and the distribution only includes them for linux x86_64.)

While technically dynamic libraries, I'm not 100% sure how externalizing them would work within the Java packaging and whatnot? (I am not a Java dev, so all I know is that these libs end up getting zipped into the .jar files that this thing needs. Which I think precludes them from being symlinked/etc the way dynamic libraries normally are? Again, not a Java dev.)

Reply:
> Hm. Let me see what things look like if I just try to entirely build signal-cli from source?

This sounds worth a try if you don't mind looking into it 👍

Reply:
My understanding is the same as @mtigas, and that separate formulae might not make sense here (due to how install happens directly into jars?)